### ISSUE TITLE
Response body starts with "while (1) {}"

### ISSUE THREAD

**Issue by pokryfka** [AdobeDocs/lightroom-partner-apis#93]
For example result of `https://lr.adobe.io/v2/health` is:

```
while (1) {}
{
    "version": "9087c3569668b1f52ca23d73e800a3146dd62da1"
}
```

Affects all queries.

**Comment by adobedeegupta** [comment]
Hi @pokryfka 

Thanks for raising this issue here.
However, this is actually not a bug, but a careful design consideration. 
Please visit [this](https://www.adobe.io/apis/creativecloud/lightroom/docs.html#!usage/calling.md) link for more details on how to parse the Lightroom API responses.

Thanks,
Deepak

**Comment by pokryfka** [comment]
@adobedeegupta 

Thank you for following on that and for the reference.

I do remove the first 12 bytes from responses which I considered a temporary hack.
I missed that it was documented.

> Returned JSON content is always prepended with a `while(1){}` clause to mitigate abuse.

To be honest I am not sure how it could prevent abuse, its still text although not a valid JSON document.

Since its documented the issue should be probably closed.
However please consider changing the value of `Content-Type` which currently is `application/json`.
The response is not a valid JSON document and it breaks all tools/libraries trying to parse it that way.

**Comment by NathanHowell** [comment]
I too find this to be an error. The body is not `application/json` as it's not valid json. I would be interested in seeing the reason why, though I'm assuming this is being done to mitigate a legacy CSRF attack.

**Comment by klefix** [comment]
just stumbled upon this as well.
I generated a nice OpenAPI Typescript Client by using the swagger.json from here https://github.com/AdobeDocs/lightroom-partner-apis/blob/master/docs/api/LightroomPartnerAPIsSpec.json

But the client could of course not parse the json, well, because it was not json.

I vote for removal of the "security feature" or change to "application/text" so we can just get text responses, apply the regex and parse the json ourselves :roll_eyes: 

Side-Note: oh, and if anyone can actually explain how this protects the API, I am very interested to know / read an article about it! In all my 10+ years of web development, this is the first time I see something like this...

**Comment by pokryfka** [comment]
> Side-Note: oh, and if anyone can actually explain how this protects the API, I am very interested to know / read an article about it! In all my 10+ years of web development, this is the first time I see something like this...

"Security through obscurity"

**Comment by patrickheinzelmann** [comment]
I also think that this "security feature" doesn't provide any additional security if you can simple remove with simple "search and replace" operation.

No other Adobe API (Creative Cloud Libraries, Adobe Analytics, Adobe Target, Adobe Experience...) has adopted this "security feature" and I haven't seen anything like that used in any other software vendor.

From my perspective, this non-working security mechanism should be removed. API versioning is a great way to support new features without breaking old integrations till they're migrated.

**Comment by TRSully** [comment]
> Hi @pokryfka
> 
> Thanks for raising this issue here. However, this is actually not a bug, but a careful design consideration. Please visit [this](https://www.adobe.io/apis/creativecloud/lightroom/docs.html#!usage/calling.md) link for more details on how to parse the Lightroom API responses.
> 
> Thanks, Deepak

Documentation link is dead. This is not a valid JSON response, this needs to be fixed or clear documentation with text/plain response added.

**Comment by alexprykhodko** [comment]
@pokryfka Another vote to fix this issue. The response is not a valid JSON document.

**Comment by jccr** [comment]
This is not intuitive 🤕

Thankfully I found this issue. I was confused because the official examples in the API docs do not reflect this: https://developer.adobe.com/lightroom/lightroom-api-docs/api/
<img width="340" alt="image" src="https://github.com/AdobeDocs/lightroom-partner-apis/assets/5132652/0a5bde4b-a296-45e9-ae48-01a297f8ea6f">

It looks like I'll need to modify the JSON decoding middleware in my project to handle this.

**Comment by klefix** [comment]
@jccr yeah, it is a shame. Also the link to the docs above now just gives me `Error 503 No healthy IP available for the backend
`... 

I think there is not much money or consideration actually spent on this API and I ditched my project idea completely now... 
I am looking for alternatives, with a proper, industry standard API...
